### ISSUE TITLE
Cursor pagination on queries sorted by custom field broken

### ISSUE THREAD

**Issue by ndigenpcc** [wp-graphql/wp-graphql#2644]
### Description

I am currently making queries from my NextJS frontend using Apollo client and queries that include sorting by an ACF field return the first page fine, however when I use fetchMore from Apollos useLazyQuery, the next page is empty. Then, when using fetchMore to go back to the previous page, it is a totally different one from the initial load.

### Steps to reproduce
1. Query partners sorting by PARTNER_TIER ex.

```
where = {...where, orderby: [{order: "DESC", field: "PARTNER_TIER"}, {order: "ASC", field: "TITLE"}]};
```

```
const GET_POSTS = gql`
query NewQuery($where: RootQueryToPartnerConnectionWhereArgs = {}) {
  partners(first: 10, where: $where) {
    nodes {
      partnerDetails {
        certifiedTrainingPartner
        childCompanyName
        contactEmail
        description
        discountAssistance
        fieldGroupName
        ordersInterface
        partnerCategory
        partnerLogo {
          altText
          sourceUrl
        }
        partnerNsInternalid
        partnerSolutionCategory
        partnerTier
        partnerWebsite
        stateProvince
      }
      id
      title
      uri
      slug
    }
    pageInfo {
      hasNextPage
      hasPreviousPage
      endCursor
      startCursor
    }
  }
}
`;
```

2. Use apollo client fetchMore to get the next page of partners, it will be blank.

3. Use apollo client fetchMore to get previous page, it will not be the original partners.

### Additional context

Correct results on first load.
<img width="1792" alt="correct initial load" src="https://user-images.githubusercontent.com/110477935/205137326-8eff7848-6fce-4450-9189-34af905c33f3.png">

Empty page after clicking next button
<img width="1788" alt="next page" src="https://user-images.githubusercontent.com/110477935/205137443-7e649ccf-482d-4861-96ad-e8e16b89bed3.png">

Trying to return to first page using previous button
<img width="1683" alt="prev page" src="https://user-images.githubusercontent.com/110477935/205137481-6f999544-1f7a-47fd-b0eb-f84bc701180d.png">

### WPGraphQL Version

1.13.4

### WordPress Version

6.0.3

### PHP Version

8.0.25

### Additional enviornment details

NextJS frontend using Apollo client

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [X] Yes
- [ ] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by jasonbahl** [comment]
@ndigenpcc can you provide more information about how you added support for the `PARTNER_TIER` field? 

The `PARTNER_TIER` field isn't a core WPGraphQL field, so it must've been filtered into the Schema and then filtered into the WPGraphQL resolvers in some way. 

We'll need more information on how you added that to the Schema and how you mapped it to the underlying Connection resolver. 

Without this information, we won't be able to reproduce or identify any issues. 

Also, was this something that was working at one point then stopped working after a plugin update? Or has it not been working at all?

**Comment by ndigenpcc** [comment]
Hi Jason, 

Here's the code I added to functions.php in my WP theme.

add_action('graphql_register_types', function () {
    $customposttype_graphql_single_name = "Partner";

    register_graphql_field('RootQueryTo' . $customposttype_graphql_single_name . 'ConnectionWhereArgs', 'partnerTier', [
        'type' => 'String',
        'description' => __('Partner Tier', 'pccdigital.wpengine.com/'),
    ]);

	register_graphql_field('RootQueryTo' . $customposttype_graphql_single_name . 'ConnectionWhereArgs', 'partnerSolutionCategory', [
        'type' => 'String',
        'description' => __('Partner Solution Category', 'pccdigital.wpengine.com/'),
    ]);

	register_graphql_field('RootQueryTo' . $customposttype_graphql_single_name . 'ConnectionWhereArgs', 'partnerCategory', [
        'type' => 'String',
        'description' => __('Partner Category', 'pccdigital.wpengine.com/'),
    ]);

	register_graphql_field('RootQueryTo' . $customposttype_graphql_single_name . 'ConnectionWhereArgs', 'stateProvince', [
        'type' => 'String',
        'description' => __('Serviced Provinces/States', 'pccdigital.wpengine.com/'),
    ]);

	register_graphql_field('RootQueryTo' . $customposttype_graphql_single_name . 'ConnectionWhereArgs', 'ordersInterface', [
        'type' => 'String',
        'description' => __('Partner services', 'pccdigital.wpengine.com/'),
    ]);

	register_graphql_field('RootQueryTo' . $customposttype_graphql_single_name . 'ConnectionWhereArgs', 'certifiedTrainingPartner', [
        'type' => 'Boolean',
        'description' => __('Training Partner Status', 'pccdigital.wpengine.com/'),
    ]);
});

add_filter( 'graphql_PostObjectsConnectionOrderbyEnum_values', function( $values ) {

	$values['PARTNER_TIER'] = [
		'value' => 'partner_tier',
		'description' => __( 'The number of likes on the post', 'wp-graphql' ),
	];

	return $values;

} );

add_filter('graphql_post_object_connection_query_args', function ($query_args, $source, $args, $context, $info) {

    $solution_type = $args['where']['partnerSolutionCategory'];
	$partner_tier = $args['where']['partnerTier'];
	$partner_category = $args['where']['partnerCategory'];
	$state_province = $args['where']['stateProvince'];
	$orders_interface = $args['where']['ordersInterface'];
	$ctp = $args['where']['certifiedTrainingPartner'];

	$query_args['meta_query'] = [
		'relation' => 'AND'
	];

	if (isset($ctp)) {
        $query_args['meta_query'][] = [
            [

                'key' => 'certified_training_partner',
                'value' => $ctp,
                'compare' => '='
            ]
        ];
    }

    if (isset($solution_type)) {
        $query_args['meta_query'][] = [
            [
                'key' => 'partner_solution_category',
                'value' => $solution_type,
                'compare' => 'LIKE'
            ]
        ];
    }

	if (isset($partner_tier)) {
        $query_args['meta_query'][] = [
            [
                'key' => 'partner_tier',
                'value' => $partner_tier,
                'compare' => '='
            ]
        ];
    }

	if (isset($partner_category)) {
        $query_args['meta_query'][] = [
            [
                'key' => 'partner_category',
                'value' => $partner_category,
                'compare' => 'LIKE'
            ]
        ];
    }

	if (isset($state_province)) {
        $query_args['meta_query'][] = [
            [
                // The key should be age, not artist_metadata
                'key' => 'state_province',
                'value' => $state_province,
                'compare' => 'LIKE'
            ]
        ];
    }

	if (isset($orders_interface)) {
        $query_args['meta_query'][] = [
            [
                // The key should be age, not artist_metadata
                'key' => 'orders_interface',
                'value' => $orders_interface,
                'compare' => 'LIKE'
            ]
        ];
    }		

    return $query_args;
}, 10, 5);

add_filter( 'graphql_post_object_connection_query_args', function( $query_args, $source, $input ) {

	if ( isset( $input['where']['orderby'] ) && is_array( $input['where']['orderby'] ) ) {

		foreach( $input['where']['orderby'] as $orderby ) {

			if ( ! isset( $orderby['field'] ) || 'partner_tier' !== $orderby['field'] ) {
				continue;
			}

			$query_args['meta_type'] = 'NUMERIC';
			$query_args['meta_key'] = 'partner_tier';
			$query_args['orderby']['meta_value_num'] = $orderby['order'];

		}

	}

	return $query_args;

}, 10, 3);

I got this code from looking at this post https://github.com/wp-graphql/wp-graphql-acf/issues/28

This was something that was never working, just developed and found the problem last week.

**Comment by krigsstev** [comment]
I have basically the same problem, though perhaps my example is simpler, so I will show my work. 

 I have an Event custom post type that is sorted on an ACF field named 'date'. I add the date field to WPGraphQL like so: 

```
add_filter('graphql_PostObjectsConnectionOrderbyEnum_values', function ($values) {

  $values['date'] = [
    'value' => 'date',
    'description' => __('The published event date', 'wp-graphql'),
  ];

  return $values;
});

add_filter('graphql_post_object_connection_query_args', function ($query_args, $source, $input) {

  if (isset($input['where']['orderby']) && is_array($input['where']['orderby'])) {

    foreach ($input['where']['orderby'] as $orderby) {

      if (!isset($orderby['field']) || 'date' !== $orderby['field']) {
        continue;
      }

      $query_args['meta_key'] = 'date';
      $query_args['orderby'] = 'meta_value_num';
      $query_args['order'] = $orderby['order'];
    }
  }

  return $query_args;
}, 10, 3);
```

Then my graphql is this: 

```
query eventBy($termId: [ID] = [18], $after: String = "", $search: String = "", $now: String = "") {
  eventsCategories(where: {termTaxonomyId: $termId}) {
    edges {
      node {
        name
        events(
          first: 2
          after: $after
          where: {dateAfter: $now, search: $search, orderby: {field: date, order: ASC}}
        ) {
          pageInfo {
            hasNextPage
            endCursor
          }
          edges {
            node {
              title
              eventsPostType {
                date
                description
                location
                ... more fields
              }
            }
          }
        }
      }
    }
  }
}
```
which returns

{
  "data": {
    "eventsCategories": {
      "edges": [
        {
          "node": {
            "name": "Arts & Culture",
            "events": {
              "pageInfo": {
                "hasNextPage": false,
                "endCursor": "YXJyYXljb25uZWN0aW9uOjY4NQ=="
              },
              "edges": [
                {
                  "node": {
                    "title": "Test Scenario 1",
                    "eventsPostType": {
                      "date": "Sat, December 24",
                      "description": null,
                      "location": "Union Square Park",
                    }
                  }
                },
                {
                  "node": {
                    "title": "Test Scenario 3",
                    "eventsPostType": {
                      "date": "Mon, December 26",
                      "description": null,
                      "location": "Union Square Park",
                    }
                  }
                },
                {
                  "node": {
                    "title": "Test Scenario 2",
                    "eventsPostType": {
                      "date": "Mon, December 26",
                      "description": null,
                      "location": "Union Square Park",
                    }
                  }
                }
              ]
            }
          }
        }
      ]
    }
  }
}

Now, there is not paging because there are only three. If we change it to first: 2, we get two, and our pageInfo is this: 

"pageInfo": {
                "hasNextPage": true,
                "endCursor": "YXJyYXljb25uZWN0aW9uOjY4Ng=="
              }

If we use the endCursor for after, we would expect to see the one remaining event, but instead we get an empty array. 

If we change it to first: 1, we get a different endCursor, which DOES in fact show us the last two events. 

Here's the problem I think: If you look at the last two events, they have the same date. In the db this is 20221226. If I change one of the dates to Dec 27, the problem is gone: I get the third item when requesting the next page by cursor. ALSO, If I leave the date the same, but change my orderBy to TITLE, then I don't have any problem paging. I've also tried adding TITLE the sort like this: orderby: [{field: date, order: ASC}, {field: TITLE, order: ASC}] but that doesn't help (nor does it actually change the sort, because I still get Test Scenario 3 before Test Scenario 2. 

So @jasonbahl  is there something wrong with my graphql_post_object_connection_query_args filter that fails when two filter values are the same where they overlap the cursor boundary? 

@ndigenpcc did you find a solution to your problem?

**Comment by krigsstev** [comment]
I tried dumping the query_args in graphql_post_object_connection_query_args before and after they are altered and I see this:

```
[23-Dec-2022 14:26:05 UTC] {
    "ignore_sticky_posts": true,
    "post_type": [
        "events"
    ],
    "no_found_rows": true,
    "post_status": "publish",
    "posts_per_page": 3,
    "graphql_cursor_compare": "<",
    "graphql_after_cursor": null,
    "graphql_before_cursor": null,
    "graphql_args": {
        "first": 2,
        "after": "",
        "where": {
            "dateAfter": "20221222",
            "orderby": [
                {
                    "field": "date",
                    "order": "ASC"
                },
                {
                    "field": "post_title",
                    "order": "ASC"
                }
            ],
            "search": ""
        }
    },
    "dateAfter": "20221222",
    "orderby": {
        "date": "ASC",
        "post_title": "ASC"
    },
    "s": "",
    "fields": "ids"
}
[23-Dec-2022 14:26:05 UTC] {
    "ignore_sticky_posts": true,
    "post_type": [
        "events"
    ],
    "no_found_rows": true,
    "post_status": "publish",
    "posts_per_page": 3,
    "graphql_cursor_compare": "<",
    "graphql_after_cursor": null,
    "graphql_before_cursor": null,
    "graphql_args": {
        "first": 2,
        "after": "",
        "where": {
            "dateAfter": "20221222",
            "orderby": [
                {
                    "field": "date",
                    "order": "ASC"
                },
                {
                    "field": "post_title",
                    "order": "ASC"
                }
            ],
            "search": ""
        }
    },
    "dateAfter": "20221222",
    "orderby": "meta_value_num",
    "s": "",
    "fields": "ids",
    "meta_key": "date",
    "order": "ASC"
}
```
I then tried a quick hard-coded altering of the added orderby, meta_key, order values to arrays like this: 

$query_args['meta_key'] = ['date', 'TITLE'];
$query_args['orderby'] = ['meta_value_num', 'meta_value'];
$query_args['order'] = [$orderby['order'], 'ASC'];

But wpgraphql throws a sql error over that. @jasonbahl how to pass multiple orderBy to the sql select statement?

**Comment by cantoute** [comment]
+1

**Comment by jasonbahl** [comment]
I believe this might be resolved by #2874

**Comment by jasonbahl** [comment]
I believe this is closed by #2882
